src/utils.js

```diff
@@ -131,9 +133,10 @@ export const randomId      = ()                   => Math.random().toString(36).
 /**
  * TODO: use generators instead? (ie. "yield")
  */
-export const iMax = (iVal, max = -Infinity) => (iVal > max ? iVal : max); 
-export const iMin = (iVal, min = +Infinity) => (iVal < min ? iVal : min);
-export const iAvg = (iVal, avg = 0, idx = 1) => (iVal + avg * (idx - 1)) / idx;
+export const iMax = (iVal, max = -Infinity) => (iVal > max ? iVal : max);
+export const iMin = (iVal, min = +Infinity) => ((iVal && iVal < min) ? iVal : min );
+export const iMinP = (iVal, min = +Infinity) => ((iVal && iVal < min && iVal > 0) ? iVal : min );
```

Comment:
a shiny new ambiguous function? (why do you need a new one?)

Reply:
There is a minimum < 0 (like for slope e.g.) and a minimum > 0 (like for speed and pace).